model.js

```diff
@@ -10,16 +10,52 @@ exports.fetchTopics = () => {
 };
 
 exports.fetchApiInformation = () => {
-    return apiInfo
-}
+  return apiInfo;
+};
 
 exports.fetchArticleWithCorrectId = (articleId) => {
-  const query = format(`SELECT * FROM articles WHERE article_id = ${articleId};`);
-  return db.query(query).then((data)=> {
-    if (data.rows.length === 0){
-      return Promise.reject({msg:'Not found!'})
+  articleId = Number(articleId);
+  if (Number.isInteger(articleId) === false) {
+    return Promise.reject({ msg: "invalid input" });
+  }
+  const query = format(
+    `SELECT * FROM articles WHERE article_id = ${articleId};`
+  );
+  return db.query(query).then((data) => {
+    if (data.rows.length === 0) {
+      return Promise.reject({ msg: "Non-existent id" });
     } else {
       return data.rows;
     }
   });
-}
\ No newline at end of file
+};
+exports.fetchArticles = (sortingKey) => {
+  let sortBy = "articles.created_at";
```

Comment:
This task doesn't require you to create a query. The sort-by should be set to default. It looks like you have started task 11 here.